From: Lee Duncan <lduncan@suse.com>
Date: Fri 15 Jul 2022 12:24:24 PM PDT
Subject: [PATCH] kABI: fix adding field to ufs_hba 
Patch-mainline: Never, kABI fix
References: git-fixes

Patch scsi-ufs-Fix-a-deadlock-in-the-error-handler.patch
added reserved_slot to struct ufs_hba, so mitigate that
by moving it to the end of the struct.
---
--- a/drivers/scsi/ufs/ufshcd.h
+++ b/drivers/scsi/ufs/ufshcd.h
@@ -766,7 +766,6 @@ struct ufs_hba {
 	u32 capabilities;
 	int nutrs;
 	int nutmrs;
-	u32 reserved_slot;
 	u32 ufs_version;
 	const struct ufs_hba_variant_ops *vops;
 	struct ufs_hba_variant_params *vps;
@@ -867,6 +866,9 @@ struct ufs_hba {
 #endif
 	u32 luns_avail;
 	bool complete_put;
+#ifndef __GENKSYMS__
+	u32 reserved_slot;
+#endif
 };
 
 /* Returns true if clocks can be gated. Otherwise false */
